p2p/src/utils/mux.rs

```diff
@@ -304,12 +303,65 @@ impl<S: Sender> GlobalSender<S> {
         recipients: Recipients<S::PublicKey>,
         payload: Bytes,
         priority: bool,
-    ) -> Result<Vec<S::PublicKey>, S::Error> {
-        let subchannel = UInt(subchannel);
-        let mut buf = BytesMut::with_capacity(subchannel.encode_size() + payload.len());
+    ) -> Result<Vec<S::PublicKey>, <S::Checked<'_> as CheckedSender>::Error> {
+        match self.check(recipients).await {
+            Ok(checked) => {
+                checked
+                    .with_subchannel(subchannel)
+                    .send(payload, priority)
+                    .await
+            }
+            Err(_) => Ok(Vec::new()),
+        }
+    }
+}
+
+impl<S: Sender> LimitedSender for GlobalSender<S> {
+    type PublicKey = S::PublicKey;
+    type Checked<'a> = CheckedGlobalSender<'a, S>;
+
+    async fn check(
+        &mut self,
+        recipients: Recipients<Self::PublicKey>,
+    ) -> Result<Self::Checked<'_>, SystemTime> {
+        self.inner
+            .check(recipients)
+            .await
+            .map(|checked| CheckedGlobalSender {
+                subchannel: 0,
```

Comment:
I almost think we should use Option<> here and assert if with_subchannel isn't called ?